presto-native-execution/presto_cpp/main/TaskResource.h

```diff
@@ -96,7 +91,7 @@ class TaskResource {
       const std::vector<std::string>& pathMatch);
 
   TaskManager& taskManager_;
-  std::shared_ptr<velox::memory::MemoryPool> pool_;
+  velox::memory::MemoryPool* pool_;
```

Comment:
So, what is this pool now being used for and why we didn't have it before?

Reply:
It is used to create constant vectors during plan conversion.